### PR TITLE
rm: linux: development-tags: add new tags

### PR DESCRIPTION
**Pull request by igoropaniuk**
Extend description of existing development tags.
Add additional development tags and details about how patches should be cherry-picked in the right way.

## Readiness

* [x] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [ ] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 354a150260e28c2ddb831c27581d087163942592 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2136/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for a080992dba4a3818bf2cc3e0027004182e9e5757 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2137/docs/artifacts/html/index.html

File: source/reference-manual/linux/development-tags.rst [hunk 1]
```diff
@@ -11,9 +11,21 @@ Foundries.io U-Boot tree::
...
+There are also exceptional tags for patches, that were cherry-picked/sent from/to SoC vendor forks:
+
+* ``[FIO from<vendor_name>]``: patches cherry-picked from ``<vendor_name>`` forked tree, for example ``[FIO fromnxp]``
+* ``[FIO to<vendor_name>]``: patches, that want to go to SoC vendor forked tree, for example ``[FIO tostm]``
+* ``[FIO to<vendor_name>-altered]``: exceptional case, when not a whole patch was cherry-picked, but rather some parts of it or a it was completely re-implemented. In this case ``-altered`` is added, for example ``[FIO fromnxp-altered]``
+
+
+.. important:: When a patch is cherry-picked, git cherry-pick command should be invoked with ``-x`` (append commit name) parameter, so
+   original commit hash is added to the new commit message (for example ``(cherry-picked from commit 1e24c2671acdbcf81207c43da39e09846f404dc3)``).
+   With a hash it is much easier to track original commit a mainline/SoC vendor tree.
```
Comment (angolini) [inline source/reference-manual/linux/development-tags.rst:31]:
```suggestion
.. important:: When a patch is cherry-picked, git cherry-pick command should be invoked with ``-x`` (append commit name) parameter, so
   the original commit hash is added to the new commit message (for example ``(cherry-picked from commit 1e24c2671acdbcf81207c43da39e09846f404dc3)``).
   With a hash, tracking the original commit in a mainline/SoC vendor tree is much easier.
```

File: source/reference-manual/linux/development-tags.rst [hunk 2]
```diff
@@ -11,9 +11,21 @@ Foundries.io U-Boot tree::
...
+* ``[FIO temphack]`` or ``[FIO hack]``: temporarily patches that keep things working for now, but need a better solution later for upstreaming
 * ``[FIO extras]``: patches pulled in for specific functionality that are useful for subscribers, but not critical
 * ``[FIO internal]``: patches needed by the LmP, not intended for upstream use
+* ``[FIO squash]``: patch should be squashed with an original patch, fixing possible issues in that patch
+
+There are also exceptional tags for patches, that were cherry-picked/sent from/to SoC vendor forks:
+
+* ``[FIO from<vendor_name>]``: patches cherry-picked from ``<vendor_name>`` forked tree, for example ``[FIO fromnxp]``
+* ``[FIO to<vendor_name>]``: patches, that want to go to SoC vendor forked tree, for example ``[FIO tostm]``
+* ``[FIO to<vendor_name>-altered]``: exceptional case, when not a whole patch was cherry-picked, but rather some parts of it or a it was completely re-implemented. In this case ``-altered`` is added, for example ``[FIO fromnxp-altered]``
```
Comment (angolini) [inline source/reference-manual/linux/development-tags.rst:26]:
```suggestion
* ``[FIO to<vendor_name>-altered]``: exceptional case, when not a whole patch was cherry-picked, but rather some parts or it was completely re-implemented. In this case ``-altered`` is added, for example ``[FIO fromnxp-altered]``
```

File: source/reference-manual/linux/development-tags.rst [hunk 3]
```diff
@@ -11,9 +11,21 @@ Foundries.io U-Boot tree::
...
-* ``[FIO temphack]``: temporarily patches that keep things working for now, but need a better solution later for upstreaming
+* ``[FIO fromtree]``: patches cherry-picked, rather than merged, from upstream (mainline)
+* ``[FIO fromlist]``: patches submitted to upstream for review (in mailing lists, peding PRs etc.), and revisions to them
+* ``[FIO toup]``: patches that want to go upstream/or have not been merged yet
+* ``[FIO temphack]`` or ``[FIO hack]``: temporarily patches that keep things working for now, but need a better solution later for upstreaming
 * ``[FIO extras]``: patches pulled in for specific functionality that are useful for subscribers, but not critical
 * ``[FIO internal]``: patches needed by the LmP, not intended for upstream use
+* ``[FIO squash]``: patch should be squashed with an original patch, fixing possible issues in that patch
+
+There are also exceptional tags for patches, that were cherry-picked/sent from/to SoC vendor forks:
```
Comment (angolini) [inline source/reference-manual/linux/development-tags.rst:22]:
```suggestion
There are also exceptional tags for patches that were cherry-picked/sent from/to SoC vendor forks:
```

File: source/reference-manual/linux/development-tags.rst [hunk 4]
```diff
@@ -11,9 +11,21 @@ Foundries.io U-Boot tree::
...
 The most common used tags through the Foundries.io repositories are:
 
-* ``[FIO fromtree]``: patches cherry-picked, rather than merged, from upstream
-* ``[FIO fromlist]``: patches submitted to upstream for review, and revisions to them
-* ``[FIO toup]``: patches that want to go upstream, but haven't yet
-* ``[FIO temphack]``: temporarily patches that keep things working for now, but need a better solution later for upstreaming
+* ``[FIO fromtree]``: patches cherry-picked, rather than merged, from upstream (mainline)
+* ``[FIO fromlist]``: patches submitted to upstream for review (in mailing lists, peding PRs etc.), and revisions to them
+* ``[FIO toup]``: patches that want to go upstream/or have not been merged yet
+* ``[FIO temphack]`` or ``[FIO hack]``: temporarily patches that keep things working for now, but need a better solution later for upstreaming
```
Comment (angolini) [inline source/reference-manual/linux/development-tags.rst:17]:
```suggestion
* ``[FIO temphack]`` or ``[FIO hack]``: temporary patches that keep things working for now but need a better solution later for upstreaming
```

File: source/reference-manual/linux/development-tags.rst [hunk 5]
```diff
@@ -11,9 +11,21 @@ Foundries.io U-Boot tree::
 
 The most common used tags through the Foundries.io repositories are:
 
-* ``[FIO fromtree]``: patches cherry-picked, rather than merged, from upstream
-* ``[FIO fromlist]``: patches submitted to upstream for review, and revisions to them
-* ``[FIO toup]``: patches that want to go upstream, but haven't yet
-* ``[FIO temphack]``: temporarily patches that keep things working for now, but need a better solution later for upstreaming
+* ``[FIO fromtree]``: patches cherry-picked, rather than merged, from upstream (mainline)
+* ``[FIO fromlist]``: patches submitted to upstream for review (in mailing lists, peding PRs etc.), and revisions to them
```
Comment (angolini) [inline source/reference-manual/linux/development-tags.rst:15]:
```suggestion
* ``[FIO fromlist]``: patches submitted to upstream for review (in mailing lists, pending PRs etc.), and revisions to them
```

**Review by angolini — APPROVED** [review]
I like it!

Only a few grammar edits.

File: source/reference-manual/linux/development-tags.rst (hunk 2 above)
Comment (kprosise) [inline source/reference-manual/linux/development-tags.rst:26]:
```suggestion
* ``[FIO to<vendor_name>-altered]``: exceptional case, when not a whole patch was cherry-picked, but rather some parts of it or a it was completely reimplemented. In this case ``-altered`` is added, for example ``[FIO fromnxp-altered]``
```

File: source/reference-manual/linux/development-tags.rst [hunk 6]
```diff
@@ -11,9 +11,21 @@ Foundries.io U-Boot tree::
...
+* ``[FIO squash]``: patch should be squashed with an original patch, fixing possible issues in that patch
+
+There are also exceptional tags for patches, that were cherry-picked/sent from/to SoC vendor forks:
+
+* ``[FIO from<vendor_name>]``: patches cherry-picked from ``<vendor_name>`` forked tree, for example ``[FIO fromnxp]``
+* ``[FIO to<vendor_name>]``: patches, that want to go to SoC vendor forked tree, for example ``[FIO tostm]``
+* ``[FIO to<vendor_name>-altered]``: exceptional case, when not a whole patch was cherry-picked, but rather some parts of it or a it was completely re-implemented. In this case ``-altered`` is added, for example ``[FIO fromnxp-altered]``
+
+
+.. important:: When a patch is cherry-picked, git cherry-pick command should be invoked with ``-x`` (append commit name) parameter, so
```
Comment (kprosise) [inline source/reference-manual/linux/development-tags.rst:29]:
```suggestion
.. important:: When a patch is cherry-picked, the Git cherry-pick command should be invoked with ``-x`` (append commit name) parameter, so
```

**Review by kprosise — APPROVED** [review]
A couple of small suggestions. Thanks!

File: source/reference-manual/linux/development-tags.rst [hunk 7]
```diff
@@ -11,9 +11,21 @@ Foundries.io U-Boot tree::
 
 The most common used tags through the Foundries.io repositories are:
 
-* ``[FIO fromtree]``: patches cherry-picked, rather than merged, from upstream
-* ``[FIO fromlist]``: patches submitted to upstream for review, and revisions to them
-* ``[FIO toup]``: patches that want to go upstream, but haven't yet
-* ``[FIO temphack]``: temporarily patches that keep things working for now, but need a better solution later for upstreaming
+* ``[FIO fromtree]``: patches cherry-picked, rather than merged, from upstream (mainline)
+* ``[FIO fromlist]``: patches submitted to upstream for review (in mailing lists, peding PRs etc.), and revisions to them
+* ``[FIO toup]``: patches that want to go upstream/or have not been merged yet
```
Comment (MrCry0) [inline source/reference-manual/linux/development-tags.rst:16]: `have not been upstreamed yet`. Patches `have not been merged` - it is `FIO fromlist` :)

File: source/reference-manual/linux/development-tags.rst [hunk 8]
```diff
@@ -11,9 +11,21 @@ Foundries.io U-Boot tree::
...
+* ``[FIO temphack]`` or ``[FIO hack]``: temporarily patches that keep things working for now, but need a better solution later for upstreaming
 * ``[FIO extras]``: patches pulled in for specific functionality that are useful for subscribers, but not critical
 * ``[FIO internal]``: patches needed by the LmP, not intended for upstream use
+* ``[FIO squash]``: patch should be squashed with an original patch, fixing possible issues in that patch
```
Comment (MrCry0) [inline source/reference-manual/linux/development-tags.rst:20]: I would add there something like: `This tag requires a commit message tag "Fixes:" filled out properly.`

**Comment by doanac** [comment]
Docs for a9302a6baf48cc9c41d0bb2f1008c5180c46a81f are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2141/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 87bce0cd3cf49fb2c3dba15c12b0b18b9306f38e are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2142/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 515fe8fcbfa824d9a6a89d8c3c1c4433db5912bc are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2145/docs/artifacts/html/index.html

**Comment by igoropaniuk** [comment]
@MrCry0 @angolini @kprosise all comments addressed, thanks!

**Comment by angolini** [comment]
it's LGTM

**Review by MrCry0 — APPROVED** [review]

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**